### ISSUE TITLE
Any methods to get the subtitle position after collision?

### ISSUE THREAD

**Issue by Rick-Xia** [Aegisub/Aegisub#209]
I have a Lua script that creates frames for subtitles. The logic behind is to duplicate each line, put it on another layer, and assign it a 'frame style'. For instance, the white frame in the screenshot below uses a frame style.
![Screen Shot 2020-12-14 at 11 05 17 AM](https://user-images.githubusercontent.com/24991836/102104510-42ff1b80-3dfc-11eb-8cf6-deebf55f20ea.png)

However, if a collision happened on those lines, their frames will be pushed a bit higher than them, as the frame styles are bigger.

I am thinking of using the line's position after collision to overwrite the position of frames using /pos. Are there any API's that I can use?

**Comment by FichteFoll** [comment]
I'm unsure what sort of "collision" you are refering to here. Can you share an example script that shows the undesired behavior with a screenshot and explain what you want instead?

**Comment by Rick-Xia** [comment]
> I'm unsure what sort of "collision" you are refering to here. Can you share an example script that shows the undesired behavior with a screenshot and explain what you want instead?

Sure, by "collision" I meant a dialogue line gets pushed up if it appears simultaneously along with another line. For instance, line 2 in this screenshot.
![Screenshot 2021-01-18 092657](https://user-images.githubusercontent.com/24991836/104927823-e4373f80-596f-11eb-856d-c4c7a28a6a90.png)

What I want is the real position of line 2. So far I tried `karaskel.preproc_line_pos` on line 2 but ended up getting the same x,y value as line 1 (pointed by the arrow at the bottom), which is the position before collision handling. I want the real line position as it gets finally rendered on video.

**Comment by FichteFoll** [comment]
Ah yes, karaskel always processes lines individually. If you know which lines will be pushed up, you could assign a separate style to them and adjust the vertical margin to where it would end up when pushed up.

**Comment by Rick-Xia** [comment]
Right, this could also work. But I noticed that the distance between dialogue lines changes (probably non-linearly) with the video resolution. Like 1080p is about 24.5px but 720p is way smaller. Do you happen to know how this distance is calculated?

**Comment by FichteFoll** [comment]
Positions should only change with script resolution (which is independant of video resolution but usually the same). Depending on how you are testing this, you may be resampling the font size when changing script resolution, which will of course result in different font heights and thus a different vertical offset from the lower line.

I don't know how exactly the margings are computed, i.e. whether it only checks what the font itself reports in its metrics or whether it also factors in the actually rendered line, but the height directly depends on the font size (in point) and the font metrics.
Note that vertical margins are specified in script pixels, just like in the `\pos` tag.

**Comment by wangqr** [comment]
If you are just trying to add borders, it can be done without getting the pos after collision.

Let's say you have inner border of 2 at layer 0 and outer border of 4 at layer 1. Now the outer border is bigger therefore pushed higher:
![img1](https://user-images.githubusercontent.com/8182182/114282362-f981ae80-9a32-11eb-927f-ae561cf393ea.png)

So we need to make some part of the text on layer 0 also have a border of 4. This can be done by simply appending `{\bord4}​` to the text (There is a [ZWSP](https://en.wikipedia.org/wiki/Zero-width_space) after the `}`):
![img2](https://user-images.githubusercontent.com/8182182/114282486-c4299080-9a33-11eb-81c4-5271a65a9f3c.png)
